### DISCUSSION TITLE
Collaborative mapping (via linked Map Parts)

### DISCUSSION THREAD

**Discussion by ghost** [OpenOrienteering/mapper#1886] [Ideas]
> **NOTE:** This is an idea on adding "Collaborative mapping" feature for organizing mapping process for remotely connected mappers.
>
> More details would be added later.

Cast @dg0yt

### Actual Behavior 

Actually *Mapper* has no way for truly "Collaborative mapping" in real-time by remotely connected mappers team.

### Idea

Main idea for implementing this would be adding feature to link external `.omap` files as "Map Part (external)".

Alternatively it could be implemented via linking `.omap` "Templates" (as read-only external map parts), that may be easily for initial implementation

### TODO

- [ ] Add "Link Map Parts" (from external `*.omap` files);
- [ ] Add "Update/Reload Map Parts" — setting option for reload "Map Parts" on defined period (in minutes) OR immediately  by clicking on "Reload Map Part" button;
  - [ ] Add "Update/Reload Templates" — same way as described in  "Update/Reload Map Parts";
- [ ] Add "Map Versioning / Revision Control" (https://github.com/OpenOrienteering/mapper/issues/1290) — would be useful, but probably unneeded for this feature (*as for initial implementation of "Collaborative mapping"*).

**Comment by jmacura** [comment]
I can imagine this would be really challenging to implement in its full potential. But a game changer, definitely! 🚀 

I understand the proposal as an ability to load a _local_ .omap file as Map Part and that file being a Dropbox/Google Drive/alike file which would be continuously synchronized with the collaborators' devices. Am I correct?

**Reply by ghost** [reply]
> Am I correct?

Yep. That is the main idea ;)

Also it should be applied for "Templates" too.

And finally merge "Map Parts" with "Templates" widget into something like "Project Browser":

- https://github.com/OpenOrienteering/mapper/issues/1121#issuecomment-654956957
